src/crypto/tests/TestChipCryptoPAL.cpp

```diff
@@ -2808,13 +2808,13 @@ TEST_F(TestChipCryptoPAL, TestVIDPID_x509Extraction)
         EXPECT_EQ(vidpid.mProductId.HasValue(), testCase.expectedPidPresent);
 
         // If present, make sure the VID matches expectation.
-        if (testCase.expectedVidPresent)
+        if (vidpid.mVendorId.HasValue())
```

Comment:
There is an expectation above of `EXPECT_EQ(vidpid.mVendorId.HasValue(), testCase.expectedVidPresent);`

If that expect passes, the logic here should be identical. If the test is hanging due to using of `Value`, we should probably use ASSERT_EQ instead?